cmd/pinshare-tray/tray.go

```diff
@@ -0,0 +1,486 @@
+	StateNotInstalled ServiceState = "NOT_INSTALLED"
+)
+
+type Tray struct {
+	// Menu items
+	menuOpenUI        *systray.MenuItem
+	menuStatus        *systray.MenuItem
+	menuIPFSStatus    *systray.MenuItem
+	menuPinShareStatus *systray.MenuItem
+	menuPeersStatus   *systray.MenuItem
+	menuSeparator1    *systray.MenuItem
+	menuStart         *systray.MenuItem
+	menuStop          *systray.MenuItem
+	menuRestart       *systray.MenuItem
+	menuSeparator2    *systray.MenuItem
+	menuSettings      *systray.MenuItem
+	menuLogs          *systray.MenuItem
+	menuAbout         *systray.MenuItem
+	menuSeparator3    *systray.MenuItem
+	menuExit          *systray.MenuItem
+
+	// State
+	serviceRunning bool
+	lastError      error
+}
+
+func NewTray() *Tray {
+	return &Tray{}
+}
+
+// BuildMenu creates the tray menu
+func (t *Tray) BuildMenu() {
+	// TODO: Re-enable when UI is ready
+	// // Open UI
+	// t.menuOpenUI = systray.AddMenuItem("Open PinShare UI", "Open the PinShare web interface")
+	//
+	// systray.AddSeparator()
+
+	// Status
+	t.menuStatus = systray.AddMenuItem("Status: Checking...", "Service status")
+	t.menuStatus.Disable()
+
+	t.menuIPFSStatus = systray.AddMenuItem("  IPFS: Unknown", "IPFS daemon status")
+	t.menuIPFSStatus.Disable()
+
+	t.menuPinShareStatus = systray.AddMenuItem("  PinShare: Unknown", "PinShare backend status")
+	t.menuPinShareStatus.Disable()
+
+	t.menuPeersStatus = systray.AddMenuItem("  Peers: Unknown", "Connected peers")
+	t.menuPeersStatus.Disable()
+
+	systray.AddSeparator()
+
+	// Service control
+	t.menuStart = systray.AddMenuItem("Start Service", "Start the PinShare service")
+	t.menuStop = systray.AddMenuItem("Stop Service", "Stop the PinShare service")
+	t.menuRestart = systray.AddMenuItem("Restart Service", "Restart the PinShare service")
+
+	systray.AddSeparator()
+
+	// Settings and logs
+	t.menuSettings = systray.AddMenuItem("Settings...", "Open settings")
+	t.menuLogs = systray.AddMenuItem("View Logs...", "Open log directory")
+
+	systray.AddSeparator()
+
+	// About
+	t.menuAbout = systray.AddMenuItem("About PinShare", "About this application")
+
+	systray.AddSeparator()
+
+	// Exit
+	t.menuExit = systray.AddMenuItem("Exit", "Exit the PinShare tray application")
+
+	// Handle menu clicks
+	go t.handleMenuClicks()
+
+	// Initial status check
+	t.updateStatus()
+}
+
+// handleMenuClicks handles menu item clicks
+func (t *Tray) handleMenuClicks() {
+	for {
+		select {
+		// TODO: Re-enable when UI is ready
+		// case <-t.menuOpenUI.ClickedCh:
+		// 	t.handleOpenUI()
+
+		case <-t.menuStart.ClickedCh:
+			t.handleStartService()
+
+		case <-t.menuStop.ClickedCh:
+			t.handleStopService()
+
+		case <-t.menuRestart.ClickedCh:
+			t.handleRestartService()
+
+		case <-t.menuSettings.ClickedCh:
+			t.handleSettings()
+
+		case <-t.menuLogs.ClickedCh:
+			t.handleViewLogs()
+
+		case <-t.menuAbout.ClickedCh:
+			t.handleAbout()
+
+		case <-t.menuExit.ClickedCh:
+			systray.Quit()
```

Comment:
I feel like we should stop the service when the user closes the tray process - my intuition (pretending to be an end-user) would be that this would be the case (end-users don't open the windows services/settings to stop things).

Reply:
I considered this but intentionally designed it the other way: the service runs independently of user login (SYSTEM context), so it should continue running when the tray closes. The tray is just a convenience UI - closing it should not affect the service. Users who want to stop the service can explicitly click Stop Service before Exit. This matches Windows service behavior expectations.